src/topography.f90

```diff
@@ -313,20 +330,27 @@ subroutine topography_number_seas(this, sea_number, number_of_seas, silent)
         im = this%nxt
         ip = 2
         if (sea(i, j) < land .and. sea(i, j) > 0) then
-          sea(i,j) = min(sea(im, j), sea(ip, j), sea(i, jm), sea(i, jp))
-          counter = counter + 1
+          new_sea = min(sea(im, j), sea(im, j), sea(ip, j), sea(i, jm), sea(i, jp))
+          if (sea(i, j) /= new_sea) then
+            sea(i, j) = new_sea
+            counter = counter + 1
+          end if
         end if
         do i = 2, this%nxt - 1
           im = i - 1
           ip = i + 1
           if (sea(i, j) < land .and. sea(i, j) > 0) then
-            !get chokes
-            choke_east = .not. (any(sea(i:ip, jp) == land) .and. any(sea(i:ip, jm) == land))
-            choke_west = .not. (any(sea(im:i, jp) == land) .and. any(sea(im:i, jm) == land))
-            choke_south = .not. (any(sea(im, jm:j) == land) .and. any(sea(ip, jm:j) == land))
-            choke_north = .not. (any(sea(im, j:jp) == land) .and. any(sea(ip, j:jp) == land))
-            new_sea = min(minval([sea(im, j), sea(ip, j), sea(i, jm), sea(i, jp)], &
-              mask=[choke_west, choke_east, choke_south, choke_north]), land)
+            if ( this%grid_type == 'C' ) then
```

Comment:
For cases like this one, I strongly prefer to use the select case construct instead of an if statement, as it's easier to extend and to catch possible mistakes (e.g., right now, if one would have `this%grid_type == 'Z'`, the code would run without any error).

Reply:
thanks, good idea

Reply:
ok done

Reply:
Could you also add a `default` clause throwing an error? Just to be defensive and future-proof this.

Reply:
good idea - done

Reply:
Thanks!